src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java

```diff
@@ -1,19 +1,56 @@
 package com.softserve.teamproject.config;
 
+import com.softserve.teamproject.service.impl.UserDetailsServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.configuration
-        .EnableWebSecurity;
-import org.springframework.security.config.annotation.web.configuration
-        .WebSecurityConfigurerAdapter;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.AuthenticationException;
 
+/**
+ * The class provides basic security configuration.
+ */
 @Configuration
 @EnableWebSecurity
-public class WebSecurityConfig  extends WebSecurityConfigurerAdapter {
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    private UserDetailsServiceImpl userDetailsService;
+
     @Override
     protected void configure(HttpSecurity http) throws Exception {
-        http
-                .authorizeRequests()
-                .antMatchers("/", "/index").permitAll();
+        // @formatter:off
+    http
+        .authorizeRequests()
+          .antMatchers("/**").authenticated().and()
+        .formLogin()
+          .defaultSuccessUrl("/welcome")
+          .usernameParameter("username")
+          .passwordParameter("password")
+          .failureUrl("/signin-error").and()
+        .logout()
+          .logoutUrl("/logout")
+          .logoutSuccessUrl("/").and()
+        .rememberMe().key("token").tokenValiditySeconds(3600);
+    // @formatter:on
```

Comment:
I've tried to use auto format without turning it back on, but it acted as if the rest of the file starting with the @formatter:off can't be formatted.  Without these comments Google Code Style keeps formatting this piece of code in one column.